src/VodafoneChannel.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace NotificationChannels\Vodafone;
+
+use GuzzleHttp\Client;
+use Illuminate\Notifications\Notification;
+use NotificationChannels\Vodafone\Exceptions\CouldNotSendNotification;
+
+class VodafoneChannel
+{
+    /** $var Client */
+    private $client;
+
+    public function __construct(VodafoneClient $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * Send the given notification.
+     *
+     * @param mixed $notifiable
+     * @param \Illuminate\Notifications\Notification $notification
+     *
+     * @throws \NotificationChannels\Vodafone\Exceptions\CouldNotSendNotification
+     */
+    public function send($notifiable, Notification $notification)
+    {
+        /* Confirm the toVodafone method exists before continuing */
+        if (! method_exists($notification, 'toVodafone')) {
+            throw CouldNotSendNotification::methodDoesNotExist();
+        }
+
+        $message = $notification->toVodafone($notifiable, $notification);
+
+        /* Check notification uses correct class for this API */
+        if (! $message instanceof VodafoneMessage) {
```

Comment:
One thing i would do here is accept a string and then just create a new `VodafoneMessage` :)